api/ssh_tunnel_proxy.ts

```diff
@@ -0,0 +1,71 @@
+import { dataform } from "@dataform/protos";
+import * as getPort from "get-port";
+import * as net from "net";
+import * as ssh2 from "ssh2";
+
+/**
+ * To connect to databases via an SSH tunnel we create a local socket on this machine
+ * on a random port, and forward traffic through an SSH tunnel to the remote SSH server
+ * where it then exits the tunnel and connect to a database on e.g. a private network.
+ */
+export class SSHTunnelProxy {
+  public static async create(
+    tunnel: dataform.ISSHTunnel,
+    destination: { host: string; port: number }
+  ) {
+    // Find a free local port for the tunnel proxy.
+    const localPort = await getPort();
+    const sshClient = new ssh2.Client();
+
+    const proxy = net.createServer(sock => {
+      sshClient.forwardOut(
+        sock.remoteAddress,
+        sock.remotePort,
+        destination.host,
+        destination.port,
+        (err, stream) => {
+          if (err) {
+            return sock.destroy();
```

Comment:
does this silently swallow the error? it seems like we'd want that bubbled up through the db adapter, somehow, if at all possible, or at least logged somewhere.

Reply:
Looks like we can feed the error into the destroy call, so caller should get this propagated back.